app/src/test/kotlin/fr/free/nrw/commons/nearby/NearbyParentFragmentPresenterTest.kt

```diff
@@ -0,0 +1,444 @@
+     */
+    @Test
+    fun testSearchViewFocusWhenBottomSheetExpanded() {
+        whenever(nearbyParentFragmentView.isListBottomSheetExpanded()).thenReturn(true)
+        nearbyPresenter.searchViewGainedFocus()
+        verify(nearbyParentFragmentView).hideBottomSheet()
+    }
+
+    /**
+     * Test if bottom details sheet gets hidden after search view gained focus
+     */
+    @Test
+    fun testSearchViewFocusWhenDetailsBottomSheetVisible() {
+        whenever(nearbyParentFragmentView.isListBottomSheetExpanded()).thenReturn(false)
+        whenever(nearbyParentFragmentView.isDetailsBottomSheetVisible()).thenReturn(true)
+        nearbyPresenter.searchViewGainedFocus()
+        verify(nearbyParentFragmentView).hideBottomDetailsSheet()
+    }
+
+    /**
+     * Test if the search is close to current location, when last location is null we expect it to
+     * return true
+     */
+    @Test
+    fun testSearchCloseToCurrentLocationNullLastLocation() {
+        whenever(nearbyParentFragmentView.getLastFocusLocation()).thenReturn(null)
+        val isClose = nearbyPresenter?.searchCloseToCurrentLocation()
+        assertTrue(isClose!!)
+    }
+
+    /**
+     * Test if the search is close to current location, when far
+     */
+    @Test
+    fun testSearchCloseToCurrentLocationWhenFar() {
+        whenever(nearbyParentFragmentView.getLastFocusLocation()).
+            thenReturn(com.mapbox.mapboxsdk.geometry.LatLng(1.0,1.0,0.0))
+        whenever(nearbyParentFragmentView.getCameraTarget()).
+                thenReturn(LatLng(2.0,1.0,0.0F))
+        //111.19 km real distance, return false if 148306.444306 >  currentLocationSearchRadius
+        NearbyController.currentLocationSearchRadius = 148306.0
+        val isClose = nearbyPresenter?.searchCloseToCurrentLocation()
+        assertFalse(isClose!!)
+    }
+
+    /**
+     * Test if the search is close to current location, when close
+     */
+    @Test
+    fun testSearchCloseToCurrentLocationWhenClose() {
+        whenever(nearbyParentFragmentView.getLastFocusLocation()).
+            thenReturn(com.mapbox.mapboxsdk.geometry.LatLng(1.0,1.0,0.0))
+        whenever(nearbyParentFragmentView.getCameraTarget()).
+            thenReturn(LatLng(2.0,1.0,0.0F))
+        //111.19 km real distance, return false if 148253.333 >  currentLocationSearchRadius
+        NearbyController.currentLocationSearchRadius = 148307.0
+        val isClose = nearbyPresenter?.searchCloseToCurrentLocation()
+        assertTrue(isClose!!)
+    }
+
+    fun expectMapAndListUpdate() {
+        nearbyPresenter.lockUnlockNearby(false)
+        whenever(nearbyParentFragmentView.isNetworkConnectionEstablished()).thenReturn(true)
+        whenever(nearbyParentFragmentView.getLastLocation()).thenReturn(latestLocation)
+    }
+
+    @Test
+    fun testSetActionListeners() {
+        nearbyPresenter.setActionListeners(any())
+        verify(nearbyParentFragmentView).setFABPlusAction(any())
```

Comment:
![image](https://user-images.githubusercontent.com/3358282/81059637-b5a03400-8ec8-11ea-8a0e-81357cbc9bc9.png)

These tests still miss a lot of this method. You can use argument captors like

```
val captor = ArgumentCaptor.for(View.OnCLickListener::class.java)
verify(nearbyParentFragmentView).setFABPlusAction(captor.capture())
captor.value.onClick(null)
```

You can also use annotations to declare an ArgumentCaptor but I wouldn't do that unless the argument captor is used in many tests